### PR TITLE
[DNM] Full prosthetic bodies.

### PR DESCRIPTION
**Pull request by Zuhayr**
DNM, isn't finished yet.

- Removed IPC.
- Removes the redundant robotic var from organs (uses status flags instead).
- Allowed groins/torsos/heads to be selected as prosthetics.
- Removed NO_BREATHE and NO_BLOOD, these are now covered by checking if the person should have lungs or a heart respectively.
- Moved a lot of NO_PAIN checks to limb level using a can_feel_pain() proc.
- Moved eye and hair species checking onto the head object.
- Removed several checks as preparation for being able to install arbitrary organs in people. No ingame effect yet ideally.
- Added a 'full prosthetic' state; if a body and all of the organs within it are prosthetic, certain variant strings will be used (copied from the IPC code).
- Renamed 'robot' type to 'drone' and 'android' type to 'robot' for clarity.
- Added specific examine text for fully prosthetic human bodies (android, gynoid, synthetic).
- revive() and restore_all_organs() will now reapply organ preferences re: prosthetics and amputation; changeling organ restore skips this.
- Added groin/chest/head icons for manufacturers.
- Makes EAL a general selectable language.
- Adjusts how radiation and vacuum overheating work for full synthetics (and partial prosthetics for overheating).
- Adds suit coolers to the autolathe.
- Changes a ton of bodypart and organ tags to constants.

Still to be done at some point:
- NO_PAIN could conceivably changed to checking if the species has a brain/human-analagous nervous system.
- NO_SCAN should really be tied to DNA somehow.
- IPC screens are not properly reimplemented.

### CODE REVIEW

**Comment by EmperorJon** [comment]
Later down the line, could Hunger tie in with the previous NO_BREATHE style stuff moving over to organs...? Like add a stomach or something and only someone with one actually needs to eat, otherwise they have a battery? Idk.

**Comment by Zuhayr** [comment]
Sure. Outside of scope for this, though.

**Comment by EmperorJon** [comment]
Yeah, cool. Was intended as a "further down the line" kinda thing, because I imagine a full body robot might be a little confusing if it eats burger :P

**Comment by zaredman** [comment]
> - Hunger is still a fact of life for full synthetics. Might fluff this as an internal reactor.

Am I the only one getting this vibe:
mrfusion_b2ttf

Serious note: Liking this. Ty much, Zuh.

**Comment by Zuhayr** [comment]
This will now also port a number of robolimb fixes from Bay.

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -878,12 +878,15 @@ var/list/admin_verbs_mentor = list(
 	if(new_fstyle)
 		M.f_style = new_fstyle
 
-	var/new_gender = alert(usr, "Please select gender.", "Character Generation", "Male", "Female")
+	var/new_gender = alert(usr, "Please select gender.", "Character Generation", "Male", "Female", "Neuter")
 	if (new_gender)
 		if(new_gender == "Male")
 			M.gender = MALE
-		else
+		else if (new_gender == "Feale")
```
Comment (Neerti) [inline code/modules/admin/admin_verbs.dm]: Typo?

File: code/modules/mob/living/carbon/carbon_powers.dm [hunk 2]
```diff
@@ -33,7 +33,7 @@
 	if(B.host_brain.ckey)
 		src << "\red <B>You send a punishing spike of psychic agony lancing into your host's brain.</B>"
 
-		if (species && (species.flags & NO_PAIN))
+		if (can_feel_pain())
```
Comment (Neerti) [inline code/modules/mob/living/carbon/carbon_powers.dm]: Should this be !can_feel_pain()?

Comment (Zuhayr) [inline code/modules/mob/living/carbon/carbon_powers.dm]: Yeah it should, well spotted.

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (Zuhayr) [inline code/modules/admin/admin_verbs.dm]: This is why I should be using constants. :(

**Comment by Zuhayr** [comment]
Testing round 1:

- Amputate full body removed left leg (could not repro)
- Setting body back to normal does not clear robogroin.
- Human synthetics do not get a synth race on examine.
- MMIs do not update icons properly.
- Names for MMIs are 'Unknown' due to mauling.
- Diona synthetics will still split apart.
- Rejuvenation gives full synthetics a meat torso.
- Organ removal surgery doesn't start.
- Blood/damage overlays use species blood colour.
- Gibber may create meat organs
- Only removing a vital organ will kill a full synthetic.

runtime error: Cannot read null.reagent_list
proc name: process (/obj/item/organ/process)
  source file: organ.dm,107
  usr: null
  src: the head (/obj/item/organ/external/head)
  src.loc: null
  call stack:
the head (/obj/item/organ/external/head): process()
the head (/obj/item/organ/external/head): process()
Unknown (/mob/living/carbon/human): handle organs()

proc name: update inv wear suit (/mob/living/carbon/human/update_inv_wear_suit)
  source file: update_icons.dm,766
runtime error: undefined variable /obj/item/weapon/holder/diona/var/blood_overlay_type

runtime error: Cannot read null.name
proc name: content (/datum/category_item/player_setup_item/player_global/pai/content)
  source file: 03_pai.dm,27
  usr: Randtheready (/mob/new_player)
  src: pAI (/datum/category_item/player_setup_item/player_global/pai)
  usr.loc: null
  call stack:
 
runtime error: Cannot execute null.Topic().
proc name: Topic (/obj/Topic)
  source file: objs.dm,23
  usr: Artificial Intelligence (/mob/living/silicon/ai)
  src: Secure Armoury Section (/obj/machinery/door/airlock/highsecurity)
  usr.loc: the processing strata (140,161,1) (/turf/simulated/floor/bluegrid)
  src.loc: the floor (112,199,1) (/turf/simulated/floor/tiled/dark)
  call stack:

**Comment by Zuhayr** [comment]
*Testing round 2*

Thank you Devildabeast for helping me test this round.

- [ ] Full prosthetic dionaea are completely fucking unkillable.
- [ ] MMI client-passing-around procs ghost the player at times (but allow them to reenter corpse).
- [ ] Removing a brain renames the brainmob/brain object to use the name of the new body, even if it has already been removed from another body.
- [ ] Something in attack code is using species blood colour instead of limb blood colour.

**Comment by Zuhayr** [comment]
**Testing round 3 (live):**

- [ ] Robot torsos have no sprite when printed from a fabricator (set to torso instead of chest).
- [ ] Full synthetics cannot *beep or *ping.
- [ ] Toxloss does not show up on the cyborg analyzer.
- [ ] Dionaea torso icon doesn't show up.
- [ ] 'gear harness' uniform item would be helpful.
- [ ] Suit cooling unit autolathe recipe produces flashlights.
- [ ] Brains can be set to 'normal' with a prosthetic body.
- [ ] You can get a meat head by resetting body, general meat resetting weirdness.
- [ ] Species/robot type can be seen even with full concealment.
- [ ] Regen to human does not reset full_prosthetic.
- [ ] Preview icon layering is busted due to token order.
- [ ] Non-synths can speak EAL.

**Comment by Zuhayr** [comment]
This is ready for merge.

**Pull request merged**